src/api/resources/classification/classification_status.py

```diff
@@ -0,0 +1,63 @@
+from django.db.models import OuterRef, Q, Subquery
+from django_filters import rest_framework as filters
+
+from ...exceptions import check_uuid
+from ...models import ClassificationStatus, CollectRecord, ObsBenthicPhotoQuadrat
+from ...permissions import ProjectDataAdminPermission, ProjectDataCollectorPermission
+from ..base import BaseAPIFilterSet, BaseAPISerializer, BaseProjectApiViewSet
+
+
+class ClassificationStatusSerializer(BaseAPISerializer):
+    class Meta:
+        model = ClassificationStatus
+        fields = ["id", "image", "status", "message", "data"]
+
+
+class ClassificationStatusFilterSet(BaseAPIFilterSet):
+    collect_record = filters.BooleanFilter(field_name="collect_record_id", method="filter_covars")
+
+    def filter_covars(self, queryset, name, value):
+        return queryset.filter(image__collect_record_id=value)
+
+    class Meta:
+        model = ClassificationStatus
+        fields = ["collect_record", "image", "status"]
+
+
+class ClassificationStatusViewSet(BaseProjectApiViewSet):
+    queryset = ClassificationStatus.objects.all()
+    serializer_class = ClassificationStatusSerializer
+    filterset_class = ClassificationStatusFilterSet
+
+    permission_classes = [
+        ProjectDataCollectorPermission,
+        ProjectDataAdminPermission,
+    ]
+
+    def get_queryset(self):
+        show_all = "showall" in self.request.query_params
+        qs = ClassificationStatus.objects.all()
+
+        if show_all is True:
+            return qs
+
+        latest_status_subquery = (
+            ClassificationStatus.objects.filter(image=OuterRef("image"))
+            .order_by("-created_on")
+            .values("id")[:1]
```

Comment:
Nitpick, but to me `[:1]` is less readable than `[0]`. Is that in order to avoid a `KeyError`, if the query returned nothing? But wouldn't we get one anyway with the brackets, and shouldn't we protect against that?

Reply:
With the `[:1]`, you get an empty list back if there's no match.

Reply:
OK yeah I'd thought that we'd need an empty queryset, not an empty list; but it looks like from the docs it is an empty queryset, so all good.